contracts/BSCValidatorSet.sol

```diff
@@ -326,18 +429,26 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
         emit failReasonWithStr("fee is larger than DUSTY_INCOMING");
         return ERROR_RELAYFEE_TOO_LARGE;
       }
-      for (uint i; i<validatorsNum; ++i) {
-        if (currentValidatorSet[i].incoming >= DUSTY_INCOMING) {
+      for (uint i; i < validatorsNum; ++i) {
+        if (isMigrated[i] == 1) {
+          if (currentValidatorSet[i].incoming != 0) {
+            directAddrs[directSize] = payable(currentValidatorSet[i].consensusAddress);
+            directAmounts[directSize] = currentValidatorSet[i].incoming;
+            isMigrated[directSize] = 1; // directSize must be less than i. so we can use directSize as index
```

Comment:
how can we mix use `isMigrated`? Seems have bug here.

Reply:
Because the new idx `directSize` is always smaller than old idx `i`. So we can overwrite `isMigrated` without error